src/app/domain/sharing/models/SharingState.ts

```diff
@@ -8,14 +8,30 @@ export enum SharingIntentStatus {
 
 export enum SharingActionType {
   SetIntentStatus = 'SET_INTENT_STATUS',
-  SetFilesToUpload = 'SET_FILES_TO_UPLOAD'
+  SetFilesToUpload = 'SET_FILES_TO_UPLOAD',
+  SetRouteToUpload = 'SET_ROUTE_TO_UPLOAD',
+  SetFlowErrored = 'SET_FLOW_ERRORED',
+  SetRecoveryState = 'SET_RECOVERY_STATE'
 }
 
 export interface SharingState {
   sharingIntentStatus: SharingIntentStatus
   filesToUpload: ReceivedFile[]
+  routeToUpload?: { href: string; slug: string }
+  errored: boolean
 }
 
 export type SharingAction =
   | { type: SharingActionType.SetIntentStatus; payload: SharingIntentStatus }
   | { type: SharingActionType.SetFilesToUpload; payload: ReceivedFile[] }
+  | {
+      type: SharingActionType.SetRouteToUpload
+      payload: { href: string; slug: string }
+    }
+  | { type: SharingActionType.SetFlowErrored; payload: boolean }
+  | { type: SharingActionType.SetRecoveryState }
+
+export interface ServiceResponse<T> {
```

Comment:
nit: Seems like you cannot have both result and error filled simultaneously, so I would reflect that in the type:

```typescript
interface ServiceResponseOk<T> {
  result: T
}

interface ServiceResponseError {
  error: string
}

export type ServiceResponse<T> = ServiceResponseOk<T> | ServiceResponseError
```

Hower this would complexify a bit type checks (exemple [here](https://github.com/cozy/cozy-flagship-app/blob/master/src/libs/clientHelpers/types.ts#L70-L99) and [here](https://github.com/cozy/cozy-flagship-app/blob/master/src/libs/clientHelpers/connectClient.ts#L47-L66)) but it ensures not type overlap is possible